tests/templates/test_tensornetworks/test_MPS.py

```diff
@@ -0,0 +1,133 @@
+            (None, None, [1,2,3,4], 1, "number of wires in each block must be larger than or equal to 2; got loc=1"),
+        ],
+    )
+    def test_exception_wrong_dim(self, block, n_params_block, wires, loc, msg_match):
+        """Verifies that an exception is raised if the number of wires or loc is incorrect."""
+        with pytest.raises(ValueError, match=msg_match):
+            MPS(wires, loc, block, n_params_block)
+
+    def circuit1(weights, wires):
+            qml.RZ(weights[0], wires = wires[0])
+            qml.RZ(weights[1], wires = wires[1])
+    @pytest.mark.parametrize(
+        ("block", "n_params_block", "wires", "loc", "weights", "msg_match"),
+        [
+            (circuit1, 2, [1,2,3,4], 2,  np.random.rand(3,3), "Weights tensor must have first dimension of length 2; got 3"),
+            (circuit1, 2, [1,2,3,4], 2,  np.random.rand(2,4), "Weights tensor must have last dimension of length 3; got 4"),
+        ],
+    )
+    def test_exception_wrong_weight_shape(self, block, n_params_block, wires, loc, weights, msg_match):
+        """Verifies that an exception is raised if the weights shape is incorrect."""
+        with pytest.raises(ValueError, match=msg_match):
+            MPS(wires, loc, block, n_params_block, weights)
+
+    # @pytest.mark.parametrize(
+    #     ("weights"),
+    #     [
+    #         ([1,2],
+    #         (circuit1, 2, [1,2,3,4], 2,  np.random.rand(2,4), "Weights tensor must have last dimension of length 3; got 4"),
+    #     ],
+    # )
+    # def test_weights_type(self, weights):
+    #     """Verifies that both lists and numpy objects are accepted as weights"""
+
+    # def test_wire_loc_match_warning():
+    #     """Tests whether a warning is raised when the number of wires doesn't correspond to loc"""
+
+    # def test_list_array_weights():
+        # """Tests whether the template accepts both lists and numpy arrays as weights"""
+
+    
+
+class TestAttributes:
+    """Tests additional methods and attributes"""
+    
+    @pytest.mark.parametrize(
+        ("n_wires", "loc", "n_params_block", "expected_shape"),
+        [
+            (4, 2, 1, (3, 1)),
+            (16, 4, 1,(7, 1)),
+        ],
+    )
+    def test_shape(self, n_wires, loc, n_params_block, expected_shape):
+        """Test that the shape method returns the correct shape of the weights tensor"""
+
+        shape = qml.MPS.shape(n_wires, loc, n_params_block)
+        assert shape == expected_shape
+
+    @pytest.mark.parametrize(
+        ("block", "n_params_block", "wires", "loc", "expected_nblock"),
+        [
+            (None, 1, [1,2,3,4], 2, 3),
+        ],
+    )
+    def test_nblocks(self, wires, loc, block, n_params_block, expected_nblock):
+        """Test that the number of blocks attribute returns the correct number"""
+
+        mps = MPS(wires, loc, block, n_params_block)
+        assert mps.n_blocks == expected_nblock
```

Comment:
I don't see a test that checks that the circuit has the shape that we want! For example, I believe all the tests would pass if we change the code in `expand`  to do something different.

We can probably check that the tape obtained from the template matches the sequence of gates we expect for a simple example. Would be good to check how this is done in other templates

Reply:
I added tests that run complete circuits and compare.